eng/pipelines/templates/steps/npm-release-task.yml

```diff
@@ -0,0 +1,54 @@
+parameters:
+  ArtifactName: ''
+  Artifact: {}
+  Registry: ''
+
+steps:
+  - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+  - ${{if eq(parameters.Registry, 'https://registry.npmjs.org/')}}:
+      - task: PowerShell@2
+        displayName: Create publish package directory
+        inputs:
+          targetType: filePath
+          filePath: eng/tools/publish-to-npm.ps1
+          arguments: -pathToArtifacts $(Package.Archive) -accessLevel "public" -tag "$(Tag)" -additionalTag "$(AdditionalTag)" -registry ${{parameters.Registry}} -npmToken $(azure-sdk-npm-token) -pipelineWorkspace $(Pipeline.Workspace)
+          pwsh: true
+        condition: succeeded()
+      - task: EsrpRelease@6
+        inputs:
+          displayName: 'Publish ${{parameters.Artifact.name}} to ESRP'
+          ConnectedServiceName: 'ESRP Release Service'
+          Intent: 'PackageDistribution'
+          ContentType: 'npm'
+          FolderLocation: $(Pipeline.Workspace)/temp/${{parameters.ArtifactName}}/${{parameters.artifact.name}}
+          Owners: $(Build.RequestedForEmail)
+          Approvers: 'azuresdk@microsoft.com'
+          ServiceEndpointUrl: 'https://api.esrp.microsoft.com'
+          MainPublisher: 'ESRPRELPACMANTEST'
+          DomainTenantId: '72f988bf-86f1-41af-91ab-2d7cd011db47'
+          ProductStateField: $(Tag)
+
+      - pwsh: |
+          . ./eng/tools/publish-to-npm.ps1
```

Comment:
Why pull in the functions here as opposed to writing this code in that script via another option or by writing another script.